internal/app/transport/watch.go

```diff
@@ -0,0 +1,54 @@
+package transport
+
+import (
+	"fmt"
+
+	gcp "github.com/envoyproxy/go-control-plane/pkg/cache/v2"
+)
+
+// Watch interface abstracts v2 and v3 watches
+type Watch interface {
+	Close()
+	GetCh() interface{}
+	Send(s interface{}) (bool, error)
+}
+
+var _ Watch = &WatchV2{}
+
+// WatchV2 is the transport object that takes care of send responses to the xds clients
+type WatchV2 struct {
+	Req *gcp.Request
```

Comment:
does this var need to be exported?

Reply:
I had to export due to a linter [error](https://github.com/golang/lint/issues/465) previously while working in EM.

Reply:
looks like linter does not complain here. making watchv2 unexported. Thanks for raising this.